src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -342,6 +354,21 @@ public override BoundNode VisitLambda(BoundLambda node)
         {
             if (_inExpressionLambda)
             {
+                var lambda = node.ExpressionSymbol as MethodSymbol;
+                if ((object)lambda != null)
+                {
+                    foreach (var p in lambda.Parameters)
+                    {
+                        if (p.RefKind != RefKind.None && p.Locations.Length != 0)
+                        {
+                            _diagnostics.Add(ErrorCode.ERR_ByRefParameterInExpressionTree, p.Locations[0]);
+                        }
+                        if (p.Type.TypeSymbol?.IsByRefLikeType == true)
```

Comment:
BoundLambda traversal doesn't traverse into the parameter syntax so we have to do additional checks here. Since the ByRefLike checks happen regardless of whether the lambda is "simple" or "parenthesized" I extracted the loop out to here.